pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -262,17 +270,94 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 	ghs.logger.Sugar().Debugf("current date: %v", currentDate)
 
 	ghs.logger.Sugar().Debug("creating a new github client")
-	client := github.NewClient(ghs.client)
 
-	repos, _, err := client.Repositories.List(ctx, ghs.cfg.GitHubOrg, nil)
+	graphqlClient := githubv4.NewClient(ghs.client)
+
+	variables := map[string]interface{}{
+		"login": githubv4.String(ghs.cfg.GitHubOrg),
+	}
+
+	// we need to check if the provided org in config.yml is a user or an organization
+	var login_query struct {
+		Organization struct {
+			Login githubv4.String
+		} `graphql:"organization(login: $login)"`
+	}
+
+	var provided_org_is_org bool
+
+	err := graphqlClient.Query(context.Background(), &login_query, variables)
 	if err != nil {
-		ghs.logger.Sugar().Errorf("Error getting repos", zap.Error(err))
+		ghs.logger.Sugar().Info("Org provided not found or org provided is user", zap.Error(err))
+		provided_org_is_org = false
+	} else {
+		provided_org_is_org = true
+	}
+	// now that we have determined if login is org or user, we can define the query
+
+	variables["repoCursor"] = (*githubv4.String)(nil)
+
+	var user_query struct {
+		User struct {
+			Repositories struct {
+				TotalCount int
+				PageInfo   struct {
+					EndCursor   githubv4.String
+					HasNextPage bool
+				}
+				Edges []RepositoryEdge
+			} `graphql:"repositories(first: 100, affiliations:OWNER, after: $repoCursor)"`
+		} `graphql:"user(login: $login)"`
+	}
+
+	var org_query struct {
+		Organization struct {
+			Repositories struct {
+				TotalCount int
+				PageInfo   struct {
+					EndCursor   githubv4.String
+					HasNextPage bool
+				}
+				Edges []RepositoryEdge
+			} `graphql:"repositories(first: 100, after: $repoCursor)"`
+		} `graphql:"organization(login: $login)"`
 	}
 
-	ghs.mb.RecordGitRepositoryCountDataPoint(now, int64(len(repos)))
+	var repos []RepositoryEdge
+	for {
+
+		// query must be dynamic to user or organization type for Graphql
+		if provided_org_is_org {
```

Comment:
I'm not sure we need to have a big conditional like this if we break things out into functions

Reply:
I will add making this its own function here: https://github.com/liatrio/liatrio-otel-collector/issues/51